### DISCUSSION TITLE
Document maiko virtual machine: opcodes, subrs, memory model, architecture, compiler considerations, etc

### DISCUSSION THREAD

**Discussion by masinter** [Interlisp/medley#1849] [Yearnings]
@nbriggs : 

> the whole "makeright ..." build process is a bit of a mess ...

>  To my mind the real problems are mostly issues with factoring the OS dependencies properly in the code (is it <strings.h> or <string.h>, do you need <time.h> or <sys/time.h> or both? Do we need <unistd.h>? Do we have <stdint.h>? Do we depend in <ioctl.h>? Do we need to use fcntl() or ioctl() to speed up the I/O for X displays? etc. etc. etc.)   If someone were, for example, trying to port the code to an ARM processor on e.g. a Raspberry Pi, what would they need to change in the configuration?  What if it were WebAsm? What about Apple Silicon?  What if we were trying to rebuild Medley 2.0 instead of 3.5?  It's not well documented!

**Comment by waywardmonkeys** [comment]
A lot of the above concerns have been evolving in the last couple of days.

**Reply by masinter** [reply]
also discussion https://github.com/orgs/Interlisp/discussions/1845

**Comment by masinter** [comment]
first step is to gather the documentation we have: comments in code, tedit files, old emails
then we can discover the gaps

**Comment by masinter** [comment]
There are some documents now in the repo under docs/internal/opcodes.tedit => medley-docs-internal-opcodes.pdf in the Medley-PDFs on the shared Internet google drive.

https://groups.google.com/d/msgid/lispcore/859A698F-30F6-4957-BB30-61C06E70A402%40gmail.com?utm_medium=email&utm_source=footer

https://gist.github.com/masinter/0c62f36fcb3750b7c108a23c61ad62c9
